scrap_location_filter/models/stock.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2021 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models
+
+
+class StockLocation(models.Model):
+    _inherit = "stock.location"
+
+    @api.model
+    def _search(
+        self,
+        args,
+        offset=0,
+        limit=None,
+        order=None,
+        count=False,
+        access_rights_uid=None,
+    ):
+        context = self._context or {}
+        if context.get("product_id") or context.get("lot_id"):
+            quants = self.env["stock.quant"].search(
+                [
+                    ("product_id", "=", context.get("product_id")),
+                    ("lot_id", "=", context.get("lot_id")),
+                ]
+            )
+            locations = quants.mapped("location_id")
+            if not locations:
+                company_id = context.get("default_company_id") or self.env.company.id
+                warehouse = self.env["stock.warehouse"].search(
+                    [("company_id", "=", company_id)], limit=1
+                )
+                if warehouse:
+                    locations = warehouse.lot_stock_id
```

Comment:
On "else" locations will not be assigned, and we are in trouble.
I say just get rid of the "if"
```suggestion
                locations = warehouse.lot_stock_id
```